pkg/compactor/bucket_compactor.go

```diff
@@ -311,16 +324,32 @@ func (c *BucketCompactor) runCompactionJob(ctx context.Context, job *Job) (shoul
 	// Once we have a plan we need to download the actual data.
 	downloadBegin := time.Now()
 
+	ignored := objstore.WithDownloadIgnoredPaths(block.MetaFilename)
 	err = concurrency.ForEachJob(ctx, len(toCompact), c.blockSyncConcurrency, func(ctx context.Context, idx int) error {
 		meta := toCompact[idx]
 
 		// Must be the same as in blocksToCompactDirs.
 		bdir := filepath.Join(subDir, meta.ULID.String())
 
-		if err := block.Download(ctx, jobLogger, c.bkt, meta.ULID, bdir); err != nil {
+		// Write meta.json to disk and then download blocks' directory w. metadata excluded
```

Comment:
Could we move this code to block.go alongside the existing Download method? Also consider if it makes sense to merge this with the existing Download method, where we write the meta to disk if it's provided, instead of fetching it again.